server/src/main/java/facultad/trendz/service/PostService.java

```diff
@@ -124,4 +122,51 @@ public void deletePost(Long postId) {
                         comment.getUser().getId()))
                 .collect(Collectors.toList());
     }
+
+    public VoteResponseDTO upvotePost(Long userId, Long postId) {
```

Comment:
you are not checking if the post has already been upvoted, so a user may add as many upvotes as they want